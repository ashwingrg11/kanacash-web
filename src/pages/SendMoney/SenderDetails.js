import React, { Component } from "react";
import { Row, Col, Container } from "react-bootstrap";

import { withRouter } from "react-router-dom";
import Header from "../../containers/layout/Header";
import { connect } from "react-redux";
import Footer from "../../containers/layout/Footer";
import SignupForm from "../../containers/user/SignupForm";
import LoginModal from "../../pages/Login/Login";
import ROUTES from "../../assets/Routes/Routes";
import PropTypes from "prop-types";

import SendMoneyHeader from "../../containers/transaction/SendMoneyHeader";
import SendingMoneyDetail from "../../containers/transaction/SendingMoneyDetail";
import CustomAlert from "../../components/Alert/CustomAlert";
import Disclaimer from "../../components/Disclaimer/Disclaimer";
class SenderDetails extends Component {
  state = {
    beneficiaryList: [],
    showLoginModal: false,
    errorMessage: "",
    variant: "",
    showAlert: false,
  };

  componentWillUnmount = () => {
    this.setState({
      beneficiaryList: [],
      showLoginModal: false,
      errorMessage: "",
      variant: "",
      showAlert: false,
    });
  };

  closeModal = () => {
    this.setState({ showAlert: false, errorMessage: "", variant: "" });
  };

  login = () => {
    this.setState({ showLoginModal: true });
  };

  closeMessage = () => {
    this.setState({ showLoginModal: false });
  };

  register = () => {
    localStorage.setItem("signUpFromTransaction", true);
    this.props.history.push(`${ROUTES.PHONE_VERIFICATION}`);
  };

  render() {
    return (
      <>
        <Header />
        <div className="shadow-line"></div>
        <section className="theme-inner">
          <Container>
            <SendMoneyHeader />
            <Row>
              <Col className="steps-content-block">
                <h2 className="sub-title">Sign Up</h2>

                <Row>
                  <Col md="6" sm="12" className="pr-md-0">
                    <SignupForm
                      successfullySignup={this.register}
                      showLogin
                      login={this.login}
                    />
                  </Col>
                  <Col
                    md="6"
                    sm="12"
                    xl={{ span: 5, offset: 1 }}
                    className="sumary-bar"
                  >
                    <SendingMoneyDetail />
                  </Col>
                </Row>
              </Col>
            </Row>
          </Container>
        </section>

        <Footer />
        {!localStorage.acceptPolicy && <Disclaimer />}

        <LoginModal
          show={this.state.showLoginModal}
          close={this.closeMessage}
        />
        <CustomAlert
          show={this.state.showAlert}
          message={this.state.errorMessage}
          variant={this.state.variant}
          close={this.closeModal}
        />
      </>
    );
  }
}
const mapStateToProps = (state) => ({
  beneficiary: state.senderDetail.beneficiaries,
  acceptPolicy: state.auth.accept_policy,
});

SenderDetails.propTypes = {
  history: PropTypes.object,
};

export default connect(mapStateToProps, null)(withRouter(SenderDetails));
