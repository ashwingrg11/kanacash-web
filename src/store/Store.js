import { createStore, combineReducers, applyMiddleware, compose } from "redux";
// import { composeWithDevTools } from 'redux-devtools-extension';
import Thunk from "redux-thunk";
import LoginReducer from "./reducers/Login";
import ForgotPasswordReducer from "./reducers/ForgotPassword";
import LoaderReducer from "./reducers/Loader";
import ModalReducer from "./reducers/Modal";
import NetworkReducer from "./reducers/Network";
import AuthReducer from "./reducers/Auth";
import ErrorReducer from "./reducers/Error";
import MiscellaneousReducer from "./reducers/MiscellaneousReducer";
import senderDetailReducers from "./reducers/SenderDetailReducer";
import TransactionReducer from "./reducers/TransactionReducer";
import SignUpReducer from "./reducers/SignUpReducer";
import AdminReducers from "./reducers/adminReducers";

const middleware = [Thunk];
const initialState = {};

const composeEnhancers =
  typeof window === "object" && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
        trace: true,
      })
    : compose;

const rootReducer = combineReducers({
  login: LoginReducer,
  forgotPassword: ForgotPasswordReducer,
  loader: LoaderReducer,
  modal: ModalReducer,
  network: NetworkReducer,
  auth: AuthReducer,
  error: ErrorReducer,
  miscellaneous: MiscellaneousReducer,
  senderDetail: senderDetailReducers,
  transaction: TransactionReducer,
  signup: SignUpReducer,
  admin: AdminReducers,
});

const store = createStore(
  rootReducer,
  initialState,
  composeEnhancers(applyMiddleware(...middleware))
);
export default store;
