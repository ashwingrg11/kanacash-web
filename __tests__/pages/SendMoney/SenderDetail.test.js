import React from "react";
import SenderDetails from "../../../src/pages/SendMoney/SenderDetails";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import { shallow } from "enzyme";
import store from "../../../src/store/Store";

import renderer from "react-test-renderer";
describe("<SenderDetails />", () => {
  it("SenderDetails renders correctly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <SenderDetails />
          </Router>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
