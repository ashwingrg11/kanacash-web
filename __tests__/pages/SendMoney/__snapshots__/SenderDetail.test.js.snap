// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SenderDetails /> SenderDetails renders correctly 1`] = `
Array [
  <header
    className="header-main"
  >
    <div
      className="container"
    >
      <nav
        className="navbar navbar-expand-lg navbar-light bg-light"
      >
        <div
          className="navbar-brand"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <img
            alt="logo"
            className="logo"
            src=""
          />
        </div>
        <button
          aria-controls="basic-navbar-nav"
          aria-label="Toggle navigation"
          className="navbar-toggler collapsed"
          onClick={[Function]}
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          aria-expanded={null}
          className="navbar-collapse collapse"
          id="basic-navbar-nav"
        >
          <div
            className="ml-md-auto navbar-nav"
            onKeyDown={[Function]}
          >
            <span
              className="btn btn-primary btn-theme btn-primary-border"
              onClick={[Function]}
            >
              Sign in
            </span>
            <button
              className="btn btn-primary btn-theme btn-primary-theme"
              onClick={[Function]}
            >
              Sign up
            </button>
          </div>
        </div>
      </nav>
    </div>
  </header>,
  <div
    className="shadow-line"
  />,
  <section
    className="theme-inner"
  >
    <div
      className="container"
    >
      <div
        className="transaction-steps row"
      >
        <div
          className="col"
        >
          <div
            className="d-flex justify-content-left align-items-center"
          >
            <div
              className="green-block payment-step"
            >
               
              <img
                alt="paymentDetailGreen"
                src=""
              />
              <span
                className="steps-title"
              >
                Payment Details
              </span>
               
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="d-flex justify-content-left align-items-center"
          >
            <div
              className="gray-block sender-step"
            >
               
              <img
                alt="SenderDetailGrey"
                src=""
              />
              <span
                className="steps-title"
              >
                Sender Details
              </span>
            </div>
             
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="d-flex justify-content-left align-items-center"
          >
             
            <div
              className="gray-block beneficiary-step"
            >
              <img
                alt="BeneficiaryDetailGrey"
                src=""
              />
              <span
                className="steps-title"
              >
                Beneficiary Details
              </span>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="d-flex justify-content-left align-items-center"
          >
             
            <div
              className="gray-block pay-info-step"
            >
              <img
                alt="paymentInfoGrey"
                src=""
              />
              <span
                className="steps-title"
              >
                Payment Information
              </span>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="d-flex justify-content-left align-items-center"
          >
            <div
              className="gray-block review-step last-step"
            >
              <img
                alt="ReviewDetailGrey"
                src=""
              />
              <span
                className="steps-title"
              >
                Review Details
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="steps-content-block col"
        >
          <h2
            className="sub-title"
          >
            Sign Up
          </h2>
          <div
            className="row"
          >
            <div
              className="pr-md-0 col-md-6 col-sm-12"
            >
              <div
                className="p-3 mb-4 col"
                style={
                  Object {
                    "backgroundColor": "#d1ecf1",
                  }
                }
              >
                Already have an account?
                <span
                  onClick={[Function]}
                  style={
                    Object {
                      "cursor": "pointer",
                      "fontWeight": "bold",
                    }
                  }
                >
                   
                  Log in
                   
                </span>
              </div>
              
              <div
                className="theme-form p-0 col"
              >
                <form
                  className="Signup-Form"
                  noValidate={true}
                  onSubmit={[Function]}
                >
                  <div
                    className="form-row"
                  >
                    <div
                      className="col-md-4 form-group"
                    >
                      <label
                        className="form-label"
                      >
                        First name
                      </label>
                      <input
                        className="form-control"
                        name="firstName"
                        onBlurCapture={[Function]}
                        onChange={[Function]}
                        placeholder="First name "
                        type="text"
                        value=""
                      />
                    </div>
                    <div
                      className="col-md-4 form-group"
                    >
                      <label
                        className="form-label"
                      >
                        Middle name
                      </label>
                      <input
                        className="form-control"
                        name="middleName"
                        onBlurCapture={[Function]}
                        onChange={[Function]}
                        placeholder="Middle name "
                        type="text"
                        value=""
                      />
                    </div>
                    <div
                      className="col-md-4 form-group"
                    >
                      <label
                        className="form-label"
                      >
                        Last name
                      </label>
                      <input
                        className="form-control"
                        name="lastName"
                        onBlurCapture={[Function]}
                        onChange={[Function]}
                        placeholder="Last name"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    className="form-row"
                  >
                    <div
                      className="form-group col-md-12"
                    >
                      <label
                        className="form-label"
                      >
                        Email address
                      </label>
                      <input
                        className="form-control"
                        name="email"
                        onBlur={[Function]}
                        onChange={[Function]}
                        placeholder="Your email address"
                        type="email"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    className="form-group"
                  >
                    <label
                      className="form-label"
                    >
                      Your state
                    </label>
                    <select
                      className="form-control"
                      name="country"
                      onBlurCapture={[Function]}
                      onChange={[Function]}
                      value=""
                    >
                      <option
                        value=""
                      >
                         Select State
                      </option>
                    </select>
                     
                  </div>
                  <div
                    className="form-row"
                  >
                    <div
                      className="form-group col"
                    >
                      <div
                        className="row"
                      >
                        <label
                          className="col-12 form-label"
                        >
                           Phone number
                        </label>
                        <div
                          className="col-md-3"
                        >
                          <select
                            className="form-control"
                            name="zip"
                            onBlurCapture={[Function]}
                            onChange={[Function]}
                            value="+1"
                          >
                            <option
                              value="+1"
                            >
                              +1
                            </option>
                          </select>
                        </div>
                        <div
                          className="col"
                        >
                          <input
                            className="form-control"
                            id="phoneNumber"
                            name="phone"
                            onBlurCapture={[Function]}
                            onChange={[Function]}
                            onInput={[Function]}
                            placeholder="Enter your phone number"
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    className="form-row"
                  >
                    <div
                      className="col-md-6 form-group"
                    >
                      <label
                        className="form-label"
                      >
                        Password
                      </label>
                      <input
                        className="form-control"
                        name="password"
                        onBlurCapture={[Function]}
                        onChange={[Function]}
                        placeholder="Set your password"
                        type="password"
                        value=""
                      />
                    </div>
                    <div
                      className="col-md-6 form-group"
                    >
                      <label
                        className="form-label"
                      >
                        Confirm Password
                      </label>
                      <input
                        className="form-control"
                        name="confirmPassword"
                        onBlurCapture={[Function]}
                        onChange={[Function]}
                        placeholder="Confirm password"
                        type="password"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    className="row"
                    style={
                      Object {
                        "marginBottom": "0px",
                      }
                    }
                  >
                    <div
                      className="col"
                    >
                      <button
                        className="bt btn-secondary btn-secondary-big btn btn-primary btn-block"
                        disabled={false}
                        type="submit"
                      >
                        Continue to next step
                      </button>
                    </div>
                  </div>
                </form>
              </div>
            </div>
            <div
              className="sumary-bar col-xl-5 col-md-6 col-sm-12 offset-xl-1"
            >
              <div
                className="row"
              >
                <div
                  className="col"
                >
                  <div
                    className="summary-card card"
                  >
                     
                    <h2
                      className="sub-title"
                    >
                      Summary
                    </h2>
                     
                    <p
                      className="small"
                    >
                       Sending money to
                    </p>
                     
                    <div
                      className="send-country"
                    >
                       
                      <span
                        className="country-flag"
                      >
                        <img
                          alt="flag"
                          src="https://cdnjs.cloudflare.com/ajax/libs/flag-icon-css/3.2.1/flags/4x3/undefined.svg"
                        />
                      </span>
                       
                      <span
                        className="sub-title"
                      />
                       
                    </div>
                    <div>
                      <hr
                        style={
                          Object {
                            "borderColor": "#212529",
                          }
                        }
                      />
                    </div>
                    <div
                      className="row"
                    >
                      <div
                        className="col"
                      >
                        <div
                          className="mt-3 row"
                        >
                          <div
                            className="col"
                          >
                            Send Amount
                             
                            <span
                              className="btn-link"
                              onClick={[Function]}
                              style={
                                Object {
                                  "color": "#1077BE",
                                  "cursor": "pointer",
                                }
                              }
                            >
                              edit
                            </span>
                          </div>
                          <div
                            className="d-flex justify-content-end col"
                            style={
                              Object {
                                "fontWeight": "bold",
                              }
                            }
                          >
                            USD 
                          </div>
                        </div>
                        <div
                          className="mt-3 row"
                        >
                          <div
                            className="col"
                          >
                            Transaction Fee
                          </div>
                          <div
                            className="d-flex justify-content-end col"
                            style={
                              Object {
                                "fontWeight": "bold",
                              }
                            }
                          >
                            USD 
                          </div>
                        </div>
                        <div
                          className="mt-3 row"
                        >
                          <div
                            className="col"
                          >
                            Total Amount
                          </div>
                          <div
                            className="d-flex justify-content-end col"
                            style={
                              Object {
                                "color": "#1077BE",
                                "fontWeight": "bold",
                              }
                            }
                          >
                            USD
                             
                            NaN
                          </div>
                        </div>
                        <div
                          className="d-flex align-items-center mt-3 xchange-rate row"
                          style={
                            Object {
                              "backgroundColor": "#d1ecf1",
                              "height": "40px",
                            }
                          }
                        >
                          <div
                            className="col"
                          >
                            Exchange Rate
                          </div>
                          <div
                            className="d-flex justify-content-end col"
                            style={
                              Object {
                                "fontWeight": "bold",
                              }
                            }
                          >
                            USD 1 =
                             
                             
                          </div>
                        </div>
                        <div
                          className="mt-3 row"
                        >
                          <div
                            className="col"
                          >
                            Receive Amount
                          </div>
                          <div
                            className="d-flex justify-content-end col"
                            style={
                              Object {
                                "fontWeight": "bold",
                              }
                            }
                          >
                             
                          </div>
                        </div>
                        <div
                          className="mt-3 row"
                        >
                          <div
                            className="col"
                          >
                            Payment Type
                          </div>
                          <div
                            className="d-flex justify-content-end col"
                            style={
                              Object {
                                "fontWeight": "bold",
                              }
                            }
                          >
                            TBD
                          </div>
                        </div>
                        <div
                          className="mt-3 row"
                        >
                          <div
                            className="col"
                          >
                            Receive Type
                          </div>
                          <div
                            className="d-flex justify-content-end col"
                            style={
                              Object {
                                "fontWeight": "bold",
                              }
                            }
                          >
                            TBD
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <footer
    className="footer-section text-center"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <div
            className="foot-nav"
          >
            <a
              aria-current="page"
              className="link active"
              href="/"
              onClick={[Function]}
              style={Object {}}
            >
              Home
            </a>
            <a
              aria-current={null}
              className="link"
              href="/about-us"
              onClick={[Function]}
            >
              About Us
            </a>
            <a
              aria-current={null}
              className="link"
              href="/terms-of-service"
              onClick={[Function]}
            >
              Terms
            </a>
            <a
              aria-current={null}
              className="link"
              href="/privacy-policy"
              onClick={[Function]}
            >
              Privacy Policy
            </a>
            <a
              aria-current={null}
              className="link"
              href="/copyright"
              onClick={[Function]}
            >
              Copyright
            </a>
            <a
              aria-current={null}
              className="link"
              href="/contact-us"
              onClick={[Function]}
            >
              Contact
            </a>
          </div>
        </div>
        <div
          className="copyright col-sm-12"
        >
          <p>
            © Golden Money Transfer Inc. 2020. All Rights Reserved.
             
            <b>
              Licensed by
            </b>
             Department of Financial Institutions, Nevada
          </p>
        </div>
      </div>
    </div>
  </footer>,
  <div
    className="disclaimer text-center"
  >
    <div
      className="container"
    >
      <div
        className="policy"
      >
         
        This website uses cookies to ensure you get the best experience on our website, as described in our
         
        <span
          className="btn-link"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
           
            privacy policy.
        </span>
      </div>
      <div
        className="policy1"
      >
        This website uses cookies,
        <span
          className="btn-link"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
           
            learn more.
        </span>
      </div>
      <div
        className="buttonDiv"
      >
        <button
          className="btn-outline-warning"
          onClick={[Function]}
          style={
            Object {
              "border": "none",
              "borderRadius": "50px",
              "color": "black",
              "padding": "5px 15px",
            }
          }
        >
          Accept
        </button>
      </div>
    </div>
  </div>,
]
`;
